### PR TITLE
Master discuss refactoring ged 27

### PR DESCRIPTION
**Pull request by ged-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/activity/activity.js [hunk 1]
```diff
@@ -39,20 +39,17 @@ export class Activity extends Component {
...
-        if (this.env.chatter) {
-            this.env.chatter.reload();
-        }
+        this.props.onUpdate();
     }
 }
 
 Object.assign(Activity, {
-    props: ["data"],
+    props: ["data", "onUpdate?"],
```
Comment (seb-odoo) [inline addons/mail/static/src/new/activity/activity.js:52]: Existence of `onUpdate` is not checked before call, it should be required.

File: addons/mail/static/src/new/views/chatter.xml [hunk 2]
```diff
@@ -111,8 +111,25 @@
             <Thread id="thread.id" t-key="thread.id" order="'desc'"/>
         </div>
     </div>
+</t>
 
-
+<t t-name="mail.activity_list" owl="1">
```
Comment (seb-odoo) [inline addons/mail/static/src/new/views/chatter.xml:116]: Not sure how much I like having separate templates that don't have their own component, but ok for now I suppose.

**Review by seb-odoo — APPROVED** [review]
Apart from above comments, the PR is ok if tests are green.

File: addons/mail/static/src/new/activity/activity.js (hunk 1 above)
Comment (ged-odoo) [reply addons/mail/static/src/new/activity/activity.js:52]: there is a default prop, so it is guaranteed to be defined. i think the activity component makes sense with an optional onUpdate prop

File: addons/mail/static/src/new/views/chatter.xml (hunk 2 above)
Comment (ged-odoo) [reply addons/mail/static/src/new/views/chatter.xml:116]: this is common in web. a template is basically an inline component without js.

**Pull request merged**